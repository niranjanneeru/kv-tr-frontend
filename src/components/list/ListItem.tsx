import { FC } from 'react';
import Status from '../status/Status';
import './style.css';
import { useNavigate } from 'react-router-dom';

type ListItemType = {
    name: string;
    id: string;
    joining_date: string;
    role: string;
    statusType: string;
    experience: number;
    department: string;
    deleteAction: (event) => void;
    editAction: (event) => void;
}

const ListItem: FC<ListItemType> = ({ name, id, joining_date, role, statusType, experience, department, deleteAction, editAction }) => {
    const navigate = useNavigate();

    function routeToEmployeeDetail() {
        navigate(`/employee/${id}`);
    }

    return (
        <tr onClick={routeToEmployeeDetail} className="list-item">
            <td>{name}</td>
            <td>{id}</td>
            <td>{joining_date}</td>
            <td>{role}</td>
            <td><Status statusType={statusType}></Status></td>
            <td>{experience} Yr</td>
            <td>{department}</td>
            <td>
                <div className='actions-item'>
                    <img className='delete-action' onClick={deleteAction} src='/assets/icons/delete.png' alt='delete-each' />
                    <img className='edit-action' onClick={editAction} src='/assets/icons/edit-list.svg' alt='edit-each' />
                </div>
            </td>
        </tr>
    );
};

export default ListItem;
