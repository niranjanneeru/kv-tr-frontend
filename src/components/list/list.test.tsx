import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import ListItem from "./ListItem";
import '@testing-library/jest-dom'

describe("Test for ListItem Component", () => {
    const mockDeleteAction = jest.fn();
    const mockEditAction = jest.fn();

    const listItemProps = {
        name: "John Doe",
        id: "123",
        joining_date: "2023-01-01",
        role: "Engineer",
        statusType: "ACTIVE",
        experience: 5,
        department: "IT",
        deleteAction: mockDeleteAction,
        editAction: mockEditAction,
    };

    it("Renders without errors", () => {
        render(
            <BrowserRouter>
                <ListItem {...listItemProps} />
            </BrowserRouter>
        );
    });

    it("Renders all data correctly", () => {
        render(
            <BrowserRouter>
                <ListItem {...listItemProps} />
            </BrowserRouter>
        );

        const nameElement = screen.getByText("John Doe");
        const idElement = screen.getByText("123");
        const roleElement = screen.getByText("Engineer");
        const experienceElement = screen.getByText("5 Yr");
        const departmentElement = screen.getByText("IT");

        expect(nameElement).toBeInTheDocument();
        expect(idElement).toBeInTheDocument();
        expect(roleElement).toBeInTheDocument();
        expect(experienceElement).toBeInTheDocument();
        expect(departmentElement).toBeInTheDocument();
    });

    it("Triggers deleteAction and editAction when buttons are clicked", () => {
        render(
            <BrowserRouter>
                <ListItem {...listItemProps} />
            </BrowserRouter>
        );

        const deleteButton = screen.getByAltText("delete-each");
        const editButton = screen.getByAltText("edit-each");

        fireEvent.click(deleteButton);
        fireEvent.click(editButton);

        expect(mockDeleteAction).toHaveBeenCalledTimes(1);
        expect(mockEditAction).toHaveBeenCalledTimes(1);
    });

    it("Navigates to employee detail page when clicked", () => {
        render(
            <BrowserRouter>
                <ListItem {...listItemProps} />
            </BrowserRouter>
        );

        const listItem = screen.getByRole("row");

        fireEvent.click(listItem);

        // Adjust this expectation based on your routing logic and URL
        expect(window.location.pathname).toBe("/employee/123");
    });
});
