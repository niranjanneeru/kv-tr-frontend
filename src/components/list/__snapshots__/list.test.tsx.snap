// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ListItem Component To Match Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="list-item"
      >
        <td>
          John Doe
        </td>
        <td>
          123
        </td>
        <td>
          2023-01-01
        </td>
        <td>
          Engineer
        </td>
        <td>
          <div
            class="status-component"
            style="background-color: rgb(211, 244, 190);"
          >
            Active
          </div>
        </td>
        <td>
          5
           Yr
        </td>
        <td>
          IT
        </td>
        <td>
          <div
            class="actions-item"
          >
            <img
              alt="delete-each"
              class="delete-action"
              src="/assets/icons/delete.png"
            />
            <img
              alt="edit-each"
              class="edit-action"
              src="/assets/icons/edit-list.svg"
            />
          </div>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="list-item"
    >
      <td>
        John Doe
      </td>
      <td>
        123
      </td>
      <td>
        2023-01-01
      </td>
      <td>
        Engineer
      </td>
      <td>
        <div
          class="status-component"
          style="background-color: rgb(211, 244, 190);"
        >
          Active
        </div>
      </td>
      <td>
        5
         Yr
      </td>
      <td>
        IT
      </td>
      <td>
        <div
          class="actions-item"
        >
          <img
            alt="delete-each"
            class="delete-action"
            src="/assets/icons/delete.png"
          />
          <img
            alt="edit-each"
            class="edit-action"
            src="/assets/icons/edit-list.svg"
          />
        </div>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
